### PR TITLE
#69 choice cache

### PR DESCRIPTION
**Pull request by Arithmomaniac**
In addition to fixing #69 and #80 , this commit:
 - Adds the first unit tests using mocks on the RSAPI provider
 - Fixes a bug that occurred when inserting a DTO with an empty list of multiple-choice values
 - Adds support for nullable single-choice fields (overwrites #94 )

### CODE REVIEW

File: Gravity/Gravity/DAL/RSAPI/RsapiDao.Get.cs [hunk 1]
```diff
@@ -110,6 +110,43 @@ private T GetHydratedDTO<T>(RDO objectRdo, ObjectFieldsDepthLevel depthLevel)
 			return dto;
 		}
 
+		private void PopulateChoices(BaseDto dto, RDO objectRdo)
+		{
+			foreach ((PropertyInfo property, RelativityObjectFieldAttribute fieldAttribute)
```
Comment (divanova-tsd) [inline Gravity/Gravity/DAL/RSAPI/RsapiDao.Get.cs]: I can't resolve lines 117 to 120. Getting two errors and have to rewrite the code in order to build:
Error 1: Foreach cannot operate on lambda expressions.
Error 2: Cannot specify constructor arguments in declaration.

File: Gravity/Gravity/Gravity.csproj [hunk 2]
```diff
@@ -76,7 +76,7 @@
     <Compile Include="DAL\RSAPI\RsapiDao.Update.cs" />
     <Compile Include="Extensions\RdoExtensions.cs" />
     <Compile Include="Extensions\TypeExtensions.cs" />
-    <Compile Include="Globals\SharedConstants.cs" />
+    <Compile Include="obj\SharedConstants.cs" />
```
Comment (divanova-tsd) [inline Gravity/Gravity/Gravity.csproj]: Getting error Source file could not be found.

File: Gravity/Gravity/DAL/RSAPI/RsapiDao.Get.cs (hunk 1 above)
Comment (Arithmomaniac) [reply Gravity/Gravity/DAL/RSAPI/RsapiDao.Get.cs]: Are you using Visual Studio 2017?

File: Gravity/Gravity/Gravity.csproj (hunk 2 above)
Comment (Arithmomaniac) [reply Gravity/Gravity/Gravity.csproj]: Fixed in newest push

File: Gravity/Gravity/DAL/RSAPI/RsapiDao.Get.cs (hunk 1 above)
Comment (divanova-tsd) [reply Gravity/Gravity/DAL/RSAPI/RsapiDao.Get.cs]: I was using VS2015...

Comment (divanova-tsd) [reply Gravity/Gravity/DAL/RSAPI/RsapiDao.Get.cs]: I have already. :)

Comment (mrobustelli) [reply Gravity/Gravity/DAL/RSAPI/RsapiDao.Get.cs]: @mrobustelli needs to get on #18!

**Comment by Arithmomaniac** [comment]
@mrobustelli Given the size of this PR, I'm leaving it for you to merge.

**Comment by mrobustelli** [comment]
OK, do you want me to merge your other pull requests as well?

**Comment by Arithmomaniac** [comment]
Yes please.

**Pull request merged**